src/app/_utils/sortEntriesByDate.ts

```diff
@@ -0,0 +1,24 @@
+import { SortType } from '@/components/SortListbox'
+
+interface SortableByDate {
+  startDate: string
+}
+
+export function sortEntriesByDate<T extends SortableByDate>(
+  entries: T[],
+  sortType: SortType,
+): T[] {
+  return entries.slice().sort((a, b) => {
```

Comment:
```suggestion
  return [...entries].sort((a, b) => {
```

I get what you are trying to do here - create a copy to avoid mutating the original array - but `slice` got me confused for a second. I wondered if you forgot to add indexes.

What do you think about my suggestion?